src/js/inverted-index.js

```diff
@@ -0,0 +1,194 @@
+/**
+ * InvertedIndex Class
+ * @class
+ */
+class Index {
+
+  /**
+   * class constructor
+   * @constructor
+   */
+  constructor() {
+    this.indexObject = {};
+    this.fileArray = [];
+  }
+
+  /**
+   * validJson
+   *
+   * Validates the file input
+   *
+   * @param {string} filename - Name of the file
+   * @param {object} file - Content of the file
+   * @return {string}
+   */
+  validJson(filename, file) {
+    if (!file.length) {
+      return 'Empty file';
+    }
+    try {
+      const jsonData = JSON.parse(file);
+      if (this.checkTitleAndText(jsonData)) {
+        return 'No title or text';
+      }
+      this.jsonData = jsonData;
+      return 'Correct file';
+    } catch (error) {
+      return 'Invalid JSON file';
+    }
+  }
+
+  /**
+   * cleanData
+   *
+   * Tokenizes data
+   *
+   * @param {string}
+   * @return {object}
+   */
+  cleanData(data) {
+    const terms = (data)
+      .toLowerCase()
+      .replace(/[^a-z0-9 ]/gi, '')
+      .split(' ');
+    return terms;
+  }
+
+  /**
+   * checkTitleAndText
+   *
+   * Checks that the array has a title and a text.
+   *
+   * @param {array}
+   * @return {bool}
+   */
+  checkTitleAndText(myArray) {
+    return myArray.some(value => !value.hasOwnProperty('title') || !value.hasOwnProperty('text'));
+  }
+
+  /**
+   * getUniqueTerms
+   *
+   * Gets the unique terms.
+   *
+   * @param {array}
+   * @return {object}
+   */
+  getUniqueTerms(terms) {
+    return terms.filter((value, index, arr) => arr.indexOf(value) === index);
+  }
+
+  /**
+   * getValues
```

Comment:
### Critical

This method name, description and param listing is very scanty and confusing, It does a massively poor job of describing what it actually does.